src/python/pants/engine/fs_test.py

```diff
@@ -1006,18 +1008,98 @@ def is_changed_snapshot() -> bool:
 
 
 # -----------------------------------------------------------------------------------------------
-# Comparison and representation of Digests
+# Native types
 # -----------------------------------------------------------------------------------------------
 
 
-def test_digest_repr() -> None:
-    assert (
-        str(Digest("f" * 64, 1))
-        == "Digest('ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff', 1)"
-    )
+@pytest.mark.parametrize("cls", [Digest, DigestPyO3])
+def test_digest_properties(cls) -> None:
+    digest = cls("a" * 64, 1000)
+    assert digest.fingerprint == "a" * 64
+    assert digest.serialized_bytes_length == 1000
+
+
+@pytest.mark.parametrize("cls", [Digest, DigestPyO3])
+def test_digest_repr(cls) -> None:
+    assert str(cls("a" * 64, 1)) == f"Digest({repr('a' * 64)}, 1)"
+
+
+@pytest.mark.parametrize("cls", [Digest, DigestPyO3])
+def test_digest_hash(cls) -> None:
+    assert hash(cls("a" * 64, 1)) == -6148914691236517206
+    assert hash(cls("b" * 64, 1)) == -4919131752989213765
+    # Note that the size bytes is not considered in the hash.
+    assert hash(cls("a" * 64, 1000)) == -6148914691236517206
+
+
+@pytest.mark.parametrize("cls", [Digest, DigestPyO3])
+def test_digest_equality(cls) -> None:
+    digest = cls("a" * 64, 1)
+    assert digest == cls("a" * 64, 1)
+    assert digest != cls("a" * 64, 1000)
+    assert digest != cls("0" * 64, 1)
+    with pytest.raises(TypeError):
+        digest < digest
+
+
+@pytest.mark.parametrize(
+    "snapshot_cls,digest_cls", [(Snapshot, Digest), (SnapshotPyO3, DigestPyO3)]
```

Comment:
Would it make sense to store the parameters in a constant to ensure consistency across all of the tests?

Reply:
Perhaps? This seems fine though: the code is temporary and only a bridge until we finish the PyO3 port. There also won't be any more tests added for `Snapshot`/`Digest`, the bindings seem pretty stable.

I suspect it's better to avoid a layer of indirection here.